torch/autograd/functional.py

```diff
@@ -0,0 +1,712 @@
+    inputs = _grad_preprocess(inputs, create_graph=create_graph, need_graph=True)
+
+    if v is not None:
+        tuple_v, v = _as_tuple(v, "v", "jvp")
+        v = _grad_preprocess(v, create_graph=create_graph, need_graph=False)
+        _validate_v(v, inputs)
+    else:
+        if len(inputs) != 1 or inputs[0].nelement() != 1:
+            raise RuntimeError("The vector v can only be None if the input to the user-provided function "
+                               "is a single Tensor with a single element.")
+
+    outputs = func(*inputs)
+    tuple_outputs, outputs = _as_tuple(outputs, "outputs of the user-provided function", "jvp")
+    _check_requires_grad(outputs, "outputs", strict=strict)
+    grad_outputs = tuple(torch.zeros_like(out, requires_grad=True) for out in outputs)
+
+    grad_inputs = _autograd_grad(outputs, inputs, grad_outputs, create_graph=True)
+    _check_requires_grad(grad_inputs, "grad_inputs", strict=strict)
+
+    grad_res = _autograd_grad(grad_inputs, grad_outputs, v, create_graph=create_graph)
+
+    jvp = tuple()
+    for i, jvp_i in enumerate(grad_res):
+        if jvp_i is None:
+            if strict:
+                raise RuntimeError("The gradient with respect to the input is independent of entry {}"
+                                   " in the grad_outputs when using the double backward trick to compute"
+                                   " forward mode gradients. This is not allowed in strict mode.".format(i))
+            jvp_i = torch.zeros_like(outputs[i])
+        else:
+            if strict and create_graph and not jvp_i.requires_grad:
+                raise RuntimeError("The jacobian of the user-provided function is independent of "
+                                   "input {}. This is not allowed in strict mode when create_graph=True.".format(i))
+        jvp += (jvp_i,)
+
+    # Cleanup objects and return them to the user
+    outputs = _grad_postprocess(outputs, create_graph)
+    jvp = _grad_postprocess(jvp, create_graph)
+
+    return _tuple_postprocess(outputs, tuple_outputs), _tuple_postprocess(jvp, tuple_outputs)
+
+
+def jacobian(func, inputs, create_graph=False, strict=False):
+    r"""Function that computes the Jacobian of a given function.
+
+    Args:
+        func (function): a Python function that takes Tensor inputs and returns
+            a tuple of Tensor or a Tensor.
+        inputs (tuple of Tensor or Tensor): inputs to the function ``func``.
+        create_graph (bool, optional): If ``True``, the Jacobian will be computed in
+            a differentiable manner. Note that when strict is ``False``, the result can not
+            require gradients or be disconnected from the inputs.
+            Defaults to ``False``.
+        strict (bool, optional): If ``True``, an error will be raised when we detect that there exists an input
+            such that all the outputs are independent of it. If ``False``, we return zeros as the vjp for
```

Comment:
"vjp" -> "jacobian"? The copy-paste here for the other docstrings might be wrong too

Reply:
Yep, bad copy paste...